tests/ui/pages/desktop/base.py

```diff
@@ -26,6 +26,15 @@ def login(self, email, password):
     def logout(self):
         self.header.click_logout()
 
+    def wait_for_page_to_load(self):
+        body = self.find_element(By.TAG_NAME, 'body')
+        self.wait.until(lambda s: 'firefox' in body.get_attribute('class'))
+
+    @property
+    def wait_for_page(self):
+        url = self.selenium.current_url
+        self.wait.until(lambda _: self.selenium.current_url != url)
```

Comment:
Is this after a click? If so, there's a geckodriver issue where it's not waiting for the page to load, which will hopefully be fixed soon. That said, it will still be worth adding an explicit wait for something on the target page. Waiting for an element to be displayed is reasonable. The wait will be specific to the page, and should be implemented in the `wait_for_page_to_load` method for the page.

Here's what we're doing in the current Addon-Tests repository for the Home page:

```python
WebDriverWait(self.selenium, self.timeout).until(
    lambda s: s.find_element(*self._promo_box_locator).size['height'] == 271)
```

This waits for the promo box on the home page to be 271 pixels tall. Unfortunately there's no magic bullet for waits, but sometimes devs can help with testability by adding something to perform last when loading a page, such as adding a class to an element.